### PR TITLE
Implementing downloading/previewing metrics and geometries

### LINKED ISSUE #12: Add data download and preview functionality
Following implementations from #11, as described in https://github.com/Urban-Analytics-Technology-Platform/popgetter-cli/issues/73#issuecomment-2373922779 this issue is for implementing download functionality for selected metrics with the following:
- Download metrics with either [`Params`](https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/blob/17069bb240ae54cb89f9baaf5f0a7d2cd858c1fb/popgetter-browser/rust_backend/src/lib.rs#L97) or [`DataRequestSpec`](https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/blob/17069bb240ae54cb89f9baaf5f0a7d2cd858c1fb/popgetter-browser/rust_backend/src/lib.rs#L111)
- Download geometries with either [`Params`](https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/blob/17069bb240ae54cb89f9baaf5f0a7d2cd858c1fb/popgetter-browser/rust_backend/src/lib.rs#L104) or [`DataRequestSpec`](https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/blob/17069bb240ae54cb89f9baaf5f0a7d2cd858c1fb/popgetter-browser/rust_backend/src/lib.rs#L125)
- Perform [client side join](https://svelte-maplibre.vercel.app/examples/data_join) on metrics and geometries
- Display result on map

### PR DESCRIPTION
**Pull request by sgreenbury**
Closes #12.

This PR is a WIP and builds on #11 with initial implementations to download metrics and geometries.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        run: |
+          cd web
+          pnpm rust-release
+          pnpm build
+
+      - name: Publish
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./web/dist/
```
Comment (dabreegster) [inline .github/workflows/ci.yml:49]: Drive-by comment cause I'm excited to try this (without having to build myself): this should be `./web/build/` for sveltekit

Comment (sgreenbury) [reply .github/workflows/ci.yml:49]: Amazing, was just trying to figure this out, thanks so much!

Comment (dabreegster) [reply .github/workflows/ci.yml]: For future debugging:
1) https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/tree/gh-pages is empty, why?
2) I went to the action logs at https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/actions/runs/11708675395/job/32610993647
3) 
image
4) 
image

I've spent hours debugging GH actions/pages nonsense before. :(  The other random trick that can sometimes help is to check if something like `wasm-pack` is generating a `.gitingore` file in a directory you're trying to publish, because the action to commit to a `gh-pages` branch will pay attention to that

Comment (sgreenbury) [reply .github/workflows/ci.yml]: Thanks a lot for the suggestions @dabreegster, they were very helpful. With the empty `gh-pages`, the build was not being committed in the publish step and it turns out `publish_dir` does not combine with the specified default `working-directory` I specified in the workflow. So updating to `publish_dir: popgetter-browser/web/build` fixed it.

**Review by andrewphilipsmith — APPROVED** [review]
Approved - A couple of comments as discussed. I will write these up and add them tomorrow
